### PR TITLE
Create 3 controllers with Swagger documentation

### PR DESCRIPTION
**Pull request by emcapone**
I've created the 3 controllers needed for my Angular project. I tested the functionality of FileUpload and I will push this to my other repo incase you'd like to see/try it. I did not put a delay on upload yet on the Angular side for testing, but I intend to do so. I did not yet implement the pets or bookmarks controllers on the Angular project. There is no database yet; everything is in memory. File uploads do save permanently.

I will start working on the pets and bookmarks implementation on my Angular project until we can talk about the status of my project and what I need to change or start next.

### CODE REVIEW

File: Controllers/BookmarksController.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+using bootcamp_api.Services;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace bootcamp_api.Controllers;
+
+[ApiController]
+[Produces("application/json")]
+[Consumes("application/json")]
+[SwaggerResponse(500, "An error occurred while processing the request", typeof(ProblemDetails))]
```
Comment (swhook52) [inline Controllers/BookmarksController.cs]: You shouldn't need a controller level swagger response. We don't need to tell the user we expect 500 errors.

File: Controllers/BookmarksController.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    public ActionResult<List<Bookmark>> GetAll() =>
+        BookmarkService.GetAll();
+
+    
+    /// <summary>
+    /// Returns a bookmark with a given id
+    /// </summary>
+    [HttpGet("{id}")]
+    [SwaggerResponse(200, "The bookmark is returned.", typeof(Bookmark))]
+    [SwaggerResponse(404, "A bookmark with the given id does not exist.", typeof(ProblemDetails))]
```
Comment (swhook52) [inline Controllers/BookmarksController.cs]: Try the `ProducesResponseType` attribute. Here's an example

```CSharp
    [ProducesResponseType(StatusCodes.Status200OK)]
    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
    [ProducesResponseType(typeof(DcamResponse), StatusCodes.Status404NotFound)]
```

File: Controllers/BookmarksController.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    public ActionResult<Bookmark> Get(int id)
+    {
+        var bookmark = BookmarkService.Get(id);
+
+        if (bookmark == null)
+            return NotFound();
+
+        return bookmark;
```
Comment (swhook52) [inline Controllers/BookmarksController.cs]: You should use the helper `Ok(bookmark)`

File: Controllers/BookmarksController.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        if (id != bookmark.Id)
+            return BadRequest();
+
+        var existingBookmark = BookmarkService.Get(id);
+        if (existingBookmark is null)
+            return NotFound();
+
+        BookmarkService.Update(bookmark);
+
+        return NoContent();
```
Comment (swhook52) [inline Controllers/BookmarksController.cs]: It's not wrong to return a No Content, but It's more common to return back the changed bookmark.

File: Controllers/FileUploadController.cs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                return Ok(new { dbPath });
+            }
+            else
+            {
+                return BadRequest();
+            }
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex}");
```
Comment (swhook52) [inline Controllers/FileUploadController.cs]: You want to protect your exception messages. We would log those messages but we would not want to send the raw text back to the client. The error could expose local file system paths, connection strings or anything else that might reveal our infrastructure. We'll talk more on this when we cover DTOs and Domain models but for now, just return a blank 500.

File: Controllers/PetsController.cs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    public IActionResult Delete(int id)
+    {
+        var pet = PetService.Get(id);
+
+        if (pet is null)
+            return NotFound();
+
+        PetService.Delete(id);
+
+        return Ok();
```
Comment (swhook52) [inline Controllers/PetsController.cs]: This should be `NoContent()` if we don't have anything to return.

**Review by swhook52 — CHANGES_REQUESTED** [review]

File: Controllers/BookmarksController.cs (hunk 1 above)
Comment (emcapone) [reply Controllers/BookmarksController.cs]: I removed these swagger responses in the latest commit.

File: Controllers/BookmarksController.cs (hunk 2 above)
Comment (emcapone) [reply Controllers/BookmarksController.cs]: I switched all of the swagger response attributes to this format in each controller.

File: Controllers/BookmarksController.cs (hunk 3 above)
Comment (emcapone) [reply Controllers/BookmarksController.cs]: This issue was fixed in both the pets and bookmarks controllers.

File: Controllers/BookmarksController.cs (hunk 4 above)
Comment (emcapone) [reply Controllers/BookmarksController.cs]: I switched Put to OK in both the pets and bookmarks controllers.

File: Controllers/PetsController.cs (hunk 6 above)
Comment (emcapone) [reply Controllers/PetsController.cs]: Delete now returns NoContent in both the pets and bookmarks controllers.

File: Controllers/FileUploadController.cs (hunk 5 above)
Comment (emcapone) [reply Controllers/FileUploadController.cs]: The 500 message is now generic and the exception message is unused until I implement logging.

**Review by swhook52 — APPROVED** [review]

**Pull request merged**